src/main/java/seedu/duke/helper/Command.java

```diff
@@ -22,143 +23,96 @@ public void viewPatient(PatientList patientList, String nric) {
-                }
-                medicineList.viewMedicine(index);
-            } catch (NumberFormatException e) {
-                ui.printParagraph("Index is out of range.");
-            }
+            medicineList.viewMedicine(parameters);
         }
     }
 
     public void addDoctor(DoctorList doctorList, String parameters) {
-        if (isNull(parameters)) {
-            ui.printNullParametersMessage();
-            ui.printAddDoctorExampleMessage();
-            return;
-        }
-        String[] addDoctorParameters = Parser.parseAddDoctor(parameters);
-        if (addDoctorParameters == null) {
-            ui.printAddPatientExampleMessage();
-        } else {
+        try {
+            String[] addDoctorParameters = Parser.parseAddDoctor(parameters);
             doctorList.add(addDoctorParameters);
-            ui.printParagraph("The doctor above has been added.");
+            UI.printParagraph("The doctor above has been added.");
+        } catch (HalpmiException e) {
+            UI.printParagraph(e.toString());
+            ui.printAddDoctorExampleMessage();
         }
     }
 
     public void deleteMedicine(MedicineList medicineList, String stringIndex) {
-        if (isNull(stringIndex)) {
-            ui.printParagraph("Parameter missing.");
-            return;
-        }
         try {
             int index = Integer.parseInt(stringIndex);
-            if (index < 1 || index > medicineList.size()) {
-                ui.printParagraph("Number is not within range.");
-                return;
-            }
-            medicineList.delete(index);
-            ui.printParagraph("The medicine record at index " + index + " has been deleted.");
-        } catch (NumberFormatException numberFormatException) {
-            ui.printParagraph("Parameter given is not a number.");
+            medicineList.deleteMedicine(index - 1);
+            UI.printParagraph("The medicine record at index " + index + " has been deleted.");
+        } catch (NumberFormatException | IndexOutOfBoundsException e) {
+            ui.printDeleteMedicineExampleMessage(medicineList);
         }
     }
 
     public void addAppointment(AppointmentList appointmentList, PatientList patientList,
                                DoctorList doctorList, String parameters) {
-        if (isNull(parameters)) {
-            ui.printNullParametersMessage();
-            return;
-        }
-        String[] addAppointmentParameters = Parser.parseAddAppointment(parameters);
-        if (addAppointmentParameters == null) {
-            ui.printAddAppointmentExampleMessage();
-        } else {
+        try {
+            String[] addAppointmentParameters = Parser.parseAddAppointment(parameters);
             appointmentList.add(addAppointmentParameters);
             UI.printParagraph("The above appointment has been added.");
+        } catch (HalpmiException e) {
+            UI.printParagraph(e.toString());
+            ui.printAddAppointmentExampleMessage();
```

Comment:
Shouldn't it be ui instead of UI? Or is there no difference?

Reply:
the print paragraph is a static method sooo i just used UI. i think actually moving on we can j change all ui into static methods maybe! tbd